sunpy/meta/meta.py

```diff
@@ -0,0 +1,189 @@
+        """Check if contains a particular metaItem"""
+
+    def __getitem__(self, key):
+        """Search for and return a metaItem with keyword"""
+
+
+    def __delitem__(self, key):
+        """Delete a metaItem with keyword"""
+
+    def __repr__(self):
+
+    def __str__(self):
+        return self.tostring()
+
+    def __eq__(self, other):
+        """
+        Two Headers are equal only if they have the exact same string
+        representation.
+        """
+        return str(self) == str(other)
+
+    def __add__(self, meta):
+        """Equivalent to concatenate"""
+
+    @property
+    def items(self):
+        """
+        The underlying data that make up this Meta; it can be
+        looked at, but it should not be modified directly.
+        """
+        return self._items
+
+    @property
+    def keywords(self):
+        """Return all of the keywords for all of the items"""
+
+    @property
+    def values(self):
+        """Returns all of the values for all of the items"""
+
+    @property
+    def comments(self):
+        """View the comments associated with each keyword."""
+
+    @property
+    def sources(self):
+        """View the sources of each of the items."""
+
+    @property
+    def _modified(self):
+        """
+        Whether or not any of the items have been modified.
+        """
+
+    @classmethod
+    def fromFITS(cls, data):
+        """Creates an MetaData from a FITS Header"""
+
+    @classmethod
+    def fromDict(cls, data):
+        """Create a MetaData from a dictionary."""
+
+    def to_string(cls):
+        """Returns a string representation of the header"""
+
+    def to_FITS(self):
+        """Returns a FITS header"""
+
+    @classmethod
+    def copy(self, strip=False):
+    """
+    Make a copy of the :class:`MetaData`.
+    """
+
+    def update(self, *args, **kwargs):
+    """Update the Header with new keyword values, updating the values of
+    existing keywords and appending new keywords otherwise; similar to
+    `dict.update`. This creates a new metaItem and adds it, keeping the old
+    one for posterity."""
+
+    def append(self, metaItems):
+        """
+        Appends a new item to the end of the Header, similar to `list.append`.
+        """
+
+    def extend(self, metaItems):
+        """
+        Appends multiple items to the header, similar to `list.extend`."""
+
+    def remove(self, keyword):
+        """
+        Removes the first item with the given keyword from the header similar
+        to `list.remove` if the Header object is treated as a list of keywords.
+
+        Parameters
+        ----------
+        keyword : str
+            The keyword of which to remove the first instance in the header
+
+        """
+        del self[self.index(keyword)]
+
+    def is_compliant(self):
+        """Check if the header is compliant (i.e. does it contain the minimum set of
+        items with keywords and by checking that each item is verified."""
+
+
+class MetaItem(object):
```

Comment:
having this class means a fundamental shift away from the
```python
>>> mymeta['spam']
'eggs'
```
approach of getting values out of a indexing operation (much like the confusion you felt about TSMD)